src/rtcModule/webrtc.cpp

```diff
@@ -2814,18 +2814,27 @@ void Session::asyncDestroy(TermCode code, const std::string& msg)
 Promise<void> Session::terminateAndDestroy(TermCode code, const std::string& msg)
 {
     if (mState == Session::kStateTerminating)
+    {
+        if (mTermCode == TermCode::kInvalid || (!isTermRetriable(code) && isTermRetriable(mTermCode)))
```

Comment:
If already terminating, there should be a termcode already set:
```suggestion
        assert(mTermCode != TermCode::kInvalid);
        if (!isTermRetriable(code) && isTermRetriable(mTermCode))
```